llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -8709,6 +8697,12 @@ VPlanPtr LoopVectorizationPlanner::tryToBuildVPlanWithVPRecipes(
                                        WithoutRuntimeCheck);
   }
   VPlanTransforms::optimizeInductionExitUsers(*Plan, IVEndValues, *PSE.getSE());
+  VPlanTransforms::runPass(VPlanTransforms::truncateToMinimalBitwidths, *Plan,
+                           CM.getMinimalBitwidths());
+  VPlanTransforms::runPass(VPlanTransforms::optimize, *Plan);
```

Comment:
`optimize` is intentionally separate from `tryToBuildVPlanWithVPRecipes`, which ideally should only take care of constructing the initial widened VPlan, to be optimize later